File: src/main/java/aginsun/kingdoms/server/items/ItemCoin.java
package aginsun.kingdoms.server.items;

import net.minecraft.client.renderer.texture.IIconRegister;
import net.minecraft.item.Item;

public final class ItemCoin extends Item
{
    public ItemCoin()
    {
        this.maxStackSize = 1;
    }

    @Override
    public void registerIcons(IIconRegister iconRegister)
    {
        this.itemIcon = iconRegister.registerIcon("taleofkingdoms:coin");
    }
}

File: src/main/java/kingdoms/server/items/ItemCoin.kt
package kingdoms.server.items

import net.minecraft.client.renderer.texture.IIconRegister
import net.minecraft.item.Item

class ItemCoin: Item()
{
    init
    {
        this.maxStackSize = 1
    }

    override fun registerIcons(iconRegister: IIconRegister)
    {
        this.itemIcon = iconRegister.registerIcon("tok:coin")
    }
}